### PR TITLE
Revenant

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ports over Revenants from TG: https://github.com/tgstation/-tg-station/pull/8842

Keep in mind, Revenant has undergone a lot of changes since the creation of that PR, so things aren't exactly the same; this PR brings them up to date to this point here: https://github.com/tgstation/-tg-station/pull/9866

What's a Revenant?

A revenant is a mini antag that can occur as an event in a shift; a dead player is selected and turned into a revenant---from here, you can go about your business; mostly absorbing the essence of dead things to gather more energy.

You can spend your energy on a few things; you can send messages to people or you can makes the light fixtures shock them, or defile areas---causing bots to become emagged.

Whenever a revenant uses an ability it gets revealed temporarily--where upon it can be attacked--of particularly powerful use is the Chaplain's Null rod, which keeps the Revenant from using any of his abilities and drains most of his energy.

Revenants can also be prevented from entering a tile by the Chaplain blessing the tile with 10 or more units of holy water.

https://github.com/ParadiseSS13/Paradise/pull/1324 (re-uploaded because I accidentally misclicked deleted this..derp).

### CODE REVIEW

**Comment by Flip-Sandwich** [comment]
Somethging something goon wraithes something something yeah they were pretty cool sure why not.

File: code/modules/mob/living/simple_animal/revenant/revenant.dm [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+/mob/living/simple_animal/revenant/proc/giveSpells()
+	if(src.mind)
+		src.mind.spell_list += new /obj/effect/proc_holder/spell/wizard/targeted/revenant_harvest
+		src.mind.spell_list += new /obj/effect/proc_holder/spell/wizard/targeted/revenant_transmit
+		src.mind.spell_list += new /obj/effect/proc_holder/spell/wizard/aoe_turf/revenant_light
+		return 1
+	return 0
+
+/mob/living/simple_animal/revenant/death()
+	if(!strikes)
```
Comment (veganzombeh) [inline code/modules/mob/living/simple_animal/revenant/revenant.dm]: Should this be `if(strikes)` instead? When strikes it zero it *should* be dying.

Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/revenant/revenant.dm]: yyyup

**Comment by Fox-McCloud** [comment]
Fairly major update to revenants. Need to test them out for any bugs---also, there is a map edit that will accompany these, but I'm mostly waiting for https://github.com/ParadiseSS13/Paradise/pull/1343 to get merged in first so as to not create conflicts; once @Xhuis finishes up his tweaks and that PR is merged in, I'll make the necessary map changes and this can be merged in.....all assuming no bugs, of course!

**Comment by Fox-McCloud** [comment]
Added in spawn points; tested out what I could--couldn't find any runtimes, glitches, or other odd behaviors with it; shoudl be merge ready if it passes review.

**Comment by Aurorablade** [comment]
*prepares to steal spawn point code for her PR*....>.>

File: code/modules/mob/living/simple_animal/revenant/revenant.dm [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+			src.mind.wipe_memory()
+			src << 'sound/effects/ghost.ogg'
+			src << "<br>"
+			src << "<span class='deadsay'><font size=3><b>You are a revenant.</b></font></span>"
+			src << "<b>Your formerly mundane spirit has been infused with alien energies and empowered into a revenant.</b>"
+			src << "<b>You are not dead, not alive, but somewhere in between. You are capable of limited interaction with both worlds.</b>"
+			src << "<b>You are invincible and invisible to everyone but other ghosts. Some abilities may change this.</b>"
+			src << "<b>To function, you are to drain the life essence from humans. This essence is a resource and will power all of your abilities.</b>"
+			src << "<b><i>You do not remember anything of your past lives, nor will you remember anything about this one after your death.</i></b>"
+			src << "<b>Be sure to read the wiki page at https://tgstation13.org/wiki/Revenant to learn more.</b>"
```
Comment (TheDZD) [inline code/modules/mob/living/simple_animal/revenant/revenant.dm]: Probably want to port over the wiki page to our wiki as well.

**Comment by Fox-McCloud** [comment]
URL/Link updated, this is ready to merge now!

**Pull request merged**